### PR TITLE
Added the possibilty to attach comment threads to entities with an annotation.

### PR DESCRIPTION
**Pull request by nicokaag**

### CODE REVIEW

**Comment by ConneXNL** [comment]
+1

**Comment by dnagirl** [comment]
yes please!

File: EventListener/DynamicRelationListener.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+{
+    /**
+     * @var Container
+     */
+    protected $container;
+
+    /**
+     * @var AnnotationReader
+     */
+    protected $reader;
```
Comment (stof) [inline EventListener/DynamicRelationListener.php]: visibility should be private

File: EventListener/DynamicRelationListener.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+ */
+class DynamicRelationListener implements EventSubscriber
+{
+    /**
+     * @var Container
+     */
+    protected $container;
+
+    /**
+     * @var AnnotationReader
```
Comment (stof) [inline EventListener/DynamicRelationListener.php]: should be Reader, not AnnotationReader.

File: EventListener/DynamicRelationListener.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+     */
+    protected $reader;
+
+    public function __construct(Container $container, Reader $reader) {
```
Comment (stof) [inline EventListener/DynamicRelationListener.php]: please don't typehint the Container class but the interface

Comment (stof) [inline EventListener/DynamicRelationListener.php]: and please don't inject the whole container when you only need to get a single parameter from it. Inject the parameter itself.

File: EventListener/DynamicRelationListener.php [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+            $commentThreadSetter = "set". ucfirst($property['property']);
+            $object->$commentThreadSetter($commentThread);
+
+            $om->persist($object);
+        }
+        $om->flush();
+    }
+
+    protected function getCommentableProperties(\ReflectionClass $reflectionProperties) {
```
Comment (stof) [inline EventListener/DynamicRelationListener.php]: should be private

File: EventListener/DynamicRelationListener.php [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            );
+        }
+    }
+
+    public function postLoad(EventArgs $args)
+    {
+        $this->addCommentThreadToObject($args);
+    }
+
+    protected function addCommentThreadToObject(EventArgs $args) {
```
Comment (stof) [inline EventListener/DynamicRelationListener.php]: should be private

File: EventListener/DynamicRelationListener.php [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getSubscribedEvents()
+    {
+        return array(
+            \Doctrine\ORM\Events::loadClassMetadata,
+            \Doctrine\ORM\Events::postLoad,
```
Comment (stof) [inline EventListener/DynamicRelationListener.php]: please don't use the ORM constants here but the strings directly. This way, you will be compatible with the MongoDB ODM too

File: Controller/ThreadController.php [hunk 7]
```diff
@@ -416,6 +416,26 @@ public function getThreadCommentsAction(Request $request, $id)
             $this->container->get('fos_comment.manager.thread')->saveThread($thread);
         }
 
+        if ($thread->getPermaLink() == '') {
```
Comment (stof) [inline Controller/ThreadController.php]: strict comparison please

**Comment by dnagirl** [comment]
Hi,

Since I'd really like this functionality, I've taken a stab at implementing the corrections required by @stof .  But since I'm a git newbie, I'm not sure whether to submit my own pull request or if there is a way to update this request.  Here's my work: https://github.com/dnagirl/FOSCommentBundle/tree/genj-master. What next?

**Comment by XWB** [comment]
@dnagirl You can simply create a new pull request.

**Comment by XWB** [comment]
Closing this one in favor of https://github.com/FriendsOfSymfony/FOSCommentBundle/pull/477

**Pull request closed without merging**